src/os/time/time.js

```diff
@@ -606,6 +623,18 @@ os.time.round = function(date, duration, opt_roundDown) {
         testDate.setUTCDate(testDate.getUTCDate() + 1);
       }
       break;
+    case 'last 7 days':
```

Comment:
I'm not certain this is the best place to handle setting up these relative ranges. `os.time.round` should support rounding up and down, and I also don't know that rounding makes sense here because we want dates to be relative to "now" versus "today at 00:00.000".

It might be better for `onDurationChanged` to have specific handling for relative dates, where the `endDate` is set to now and `startDate` is set to the end date minus the appropriate duration. 

One issue with this is it would set the range at the time the user changes the duration, but would remain fixed to that specific range. If someone is using the application for several hours and later makes a query, they will be querying data relative to when they changed the duration not the current time. We may need to handle this within the timeline controller instead of setting a static range once within the date control.